File: java/203. Remove Linked List Elements/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {

        ListNode preHead = new ListNode(-1 , head);
        ListNode prev = preHead;
        ListNode current = head;

        while (current != null) {
            if (current.val == val) {
                prev.next = current.next;
            } else {
                prev = prev.next;
            }
            current = current.next;
        }
        return preHead.next;
    }
}

File: python/203. Remove Linked List Elements/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:

        preHead = ListNode(-1, head)
        prev = preHead
        current = head

        while current:
            if current.val == val:
                prev.next = current.next
            else:
                prev = prev.next
            current = current.next

        return preHead.next